### PR TITLE
ISPN-13873 Handle Accept header to serialize the events

### PR DESCRIPTION
**Pull request by jabolina**
https://issues.redhat.com/browse/ISPN-13873

* Created an `EventLog` serializer using the
`ConfigurationSerializer` interface.

I am using the stuff that we already have to serialize configuration, since that can handle the `EventLog` object, but pretty much all of the serialization classes have `Configuration` in the name.

Now we have:

```json
{
  "log" : {
    "instant" : "2022-07-26T11:33:08.666580Z",
    "category" : "LIFECYCLE",
    "content" : {
      "level" : "INFO",
      "message" : "ISPN100010: Finished rebalance with members [jbolina-34174, jbolina-58560], topology id 2"
    },
    "additional" : {
      "context" : "caching-listen",
      "scope" : "jbolina-58560",
      "detail" : null,
      "who" : null
    }
  }
}
```

```yaml
log: 
  instant: "2022-07-26T11:11:22.912906Z"
  category: "LIFECYCLE"
  content: 
    level: "INFO"
    message: "ISPN100002: Starting rebalance with members [jbolina-9736, jbolina-46792], phase READ_OLD_WRITE_ALL, topology id 2"
  additional: 
    context: "caching-listen"
    scope: "jbolina-9736"
    detail: ~
    who: ~
```

```xml
<?xml version="1.0"?>
<log instant="2022-07-26T11:31:07.318201Z" category="LIFECYCLE">
    <content level="INFO" message="ISPN100002: Starting rebalance with members [jbolina-23979, jbolina-17669], phase READ_OLD_WRITE_ALL, topology id 2"/>
    <additional context="caching-listen" scope="jbolina-23979" detail="" who=""/>
</log>
```

Want to check with you all first. This is changes the object format for the log, currently is:

```json
{
  "message":"ISPN100002: Starting rebalance with members [jbolina-29050, jbolina-38947], phase READ_OLD_WRITE_ALL, topology id 2",
  "category":"LIFECYCLE",
  "level":"INFO",
  "time":1658846192341,
  "owner":"",
  "context":"caching-listen",
  "scope":"jbolina-38947"
}
```

Do we keep the format that is currently used or it is OK to change it?

### CODE REVIEW

**Comment by ryanemerson** [comment]
> Do we keep the format that is currently used or it is OK to change it?

It's OK to change it as the current format was never in a stable release, only DevXX.

It would be good if we could include an integration test consuming the events in the various formats via `server/tests`.

File: core/src/main/java/org/infinispan/util/logging/events/EventLogSerializer.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      writer.writeStartElement("log");
+      writer.writeAttribute("instant", event.getWhen().toString());
+      writer.writeAttribute("category", event.getCategory().name());
+
+      writer.writeStartElement("content");
+      writer.writeAttribute("level", event.getLevel().name());
+      writer.writeAttribute("message", event.getMessage());
+      writer.writeEndElement();
+
+      writer.writeStartElement("additional");
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/util/logging/events/EventLogSerializer.java:24]: Do we need the `additional` element or should we just include this as part of the `content` element? The `EventLog#getDetail` field strikes me as something that is definitely part of the "content". The other fields could be classed as meta data.

Comment (jabolina) [reply core/src/main/java/org/infinispan/util/logging/events/EventLogSerializer.java:24]: I've added the `additional` so we could make a distinction of some kind between what was the message and what was just meta data. But I defined these arbitrarily, I am going to update it.

**Comment by jabolina** [comment]
> However, tt would be good if we could include an integration test consuming the events in the various formats via `server/tests`.

Added a simpler version for now, where we verify specifically the `log` events. What do you think? When we publish more events in the listener we update the tests to be more generic on the content validation.

**Comment by jabolina** [comment]
Oooops, I need to update the docs to reference the headers.

File: core/src/main/java/org/infinispan/util/logging/events/EventLogSerializer.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      writer.writeAttribute("instant", event.getWhen().toString());
+      writer.writeAttribute("category", event.getCategory().name());
+
+      writer.writeStartElement("content");
+      writer.writeAttribute("level", event.getLevel().name());
+      writer.writeAttribute("message", event.getMessage());
+      writer.writeAttribute("detail", unwrap(event.getDetail()));
+      writer.writeEndElement();
+
+      writer.writeStartElement("additional");
```
Comment (ryanemerson) [inline core/src/main/java/org/infinispan/util/logging/events/EventLogSerializer.java:25]: IMO this should be called "meta" or "metadata" to make it's purpose more clear.

I also think that the "instant" attribute should be a child of this element, so we have three top level elements of:

```yaml
log: 
  category:
  content:
  meta:
```

**Comment by ryanemerson** [comment]
Just one comment and then I think we're good to go.

**Comment by ryanemerson** [comment]
Thanks @jabolina

**Pull request merged**